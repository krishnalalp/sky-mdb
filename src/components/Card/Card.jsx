import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import constants from "../../helpers/constants";

import "./Card.scss";

const { TYPES } = constants;

export function Card({ type, data }) {
  const renderItem = (itemType, id, title, image) => (
    <Link to={`/${itemType}/${id}`}>
      <div className="card">
        <div
          className="card-cover"
          style={
            image
              ? {
                  backgroundImage: `url('${constants.IMAGE_BASE_URL}${image}')`,
                }
              : { backgroundImage: `url('${constants.DEFAULT_IMAGE}')` }
          }
        />
        <div className="card-body">
          <div className="card-detail">
            <div className="card-title">{title}</div>
          </div>
        </div>
      </div>
    </Link>
  );

  const renderSearchItem = () => {
    const { title, name, poster_path, profile_path, id } = data;
    if (type === TYPES.CAST) {
      return renderItem(TYPES.CAST, id, name, profile_path);
    }
    if (type === TYPES.TV) {
      return renderItem(TYPES.TV, id, name, poster_path);
    }
    return renderItem(TYPES.MOVIE, id, title, poster_path);
  };

  return <div className="search-item">{renderSearchItem()}</div>;
}

Card.propTypes = {
  type: PropTypes.string.isRequired,
  data: PropTypes.object.isRequired,
};
