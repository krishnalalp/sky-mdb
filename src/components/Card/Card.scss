.search-item {
  padding-right: 20px;
  width: 16.25%;
  min-width: 300px;
  margin-bottom: 20px;
}

.card {
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  color: rgba(0, 0, 0, 0.85);
  font-size: 14px;
  font-variant: tabular-nums;
  line-height: 1.5715;
  list-style: none;
  -webkit-font-feature-settings: "tnum";
  font-feature-settings: "tnum", "tnum";
  position: relative;
  background: #fff;
  border-radius: 2px;
  cursor: pointer;
  -webkit-transition: border-color 0.3s, -webkit-box-shadow 0.3s;
  transition: border-color 0.3s, -webkit-box-shadow 0.3s;
  transition: box-shadow 0.3s, border-color 0.3s;
  transition: box-shadow 0.3s, border-color 0.3s, -webkit-box-shadow 0.3s;

  &:hover {
    transform: scale(1.15);
    z-index: 2;
    transition: all 0.1s linear;
    -webkit-box-shadow: 0px 11px 28px -2px #000;
    -moz-box-shadow: 0px 11px 28px -2px #000;
    box-shadow: 0px 11px 28px -2px #000;

    .card-title {
      white-space: normal;
    }
  }
}

.card-cover {
  margin-top: -1px;
  margin-right: -1px;
  margin-left: -1px;
  height: 15vw;
  min-height: 180px;
  background-repeat: no-repeat;
  background-size: cover;
}

.card-body {
  padding: 24px;
  background: #202020;
  color: #fff;
}

.card-detail {
  overflow: hidden;
}

.card-title {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}
