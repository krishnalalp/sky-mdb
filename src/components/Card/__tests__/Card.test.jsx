import React from "react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import renderer from "react-test-renderer";
import { Card } from "../Card";
import { cardData } from "./mockData";

test("Card component rendered", () => {
  const history = createMemoryHistory();
  const { type, data } = cardData;
  const component = renderer.create(
    <Router history={history}>
      <Card type={type} data={data} />,
    </Router>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
