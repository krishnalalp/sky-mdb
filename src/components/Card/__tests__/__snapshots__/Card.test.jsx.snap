// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component rendered 1`] = `
Array [
  <div
    className="search-item"
  >
    <a
      href="/movie/724989"
      onClick={[Function]}
    >
      <div
        className="card"
      >
        <div
          className="card-cover"
          style={
            Object {
              "backgroundImage": "url('http://image.tmdb.org/t/p/w500/ugZW8ocsrfgI95pnQ7wrmKDxIe.jpg')",
            }
          }
        />
        <div
          className="card-body"
        >
          <div
            className="card-detail"
          >
            <div
              className="card-title"
            >
              Hard Kill
            </div>
          </div>
        </div>
      </div>
    </a>
  </div>,
  ",",
]
`;
